embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt

```diff
@@ -430,4 +464,14 @@ internal class ModuleInitBootstrapper(
     }
 
     private fun isInitialized(): Boolean = asyncInitTask.get() != null
+
+    private fun <T> init(module: KClass<*>, provider: Provider<T>): T =
+        Systrace.traceSynchronous("${toSectionName(module)}-init") { provider() }
+
+    private fun <T> postInit(module: KClass<*>, code: () -> T): T =
+        Systrace.traceSynchronous("${toSectionName(module)}-post-init") { code() }
+
+    // This is called twice for each input - memoizing/caching is not worth the hassle
+    private fun toSectionName(klass: KClass<*>): String =
+        klass.simpleName?.removeSuffix("Module")?.toLowerCase(Locale.ENGLISH) ?: "module"
```

Comment:
Note: we already have it in our proguard files to not obfuscate our classes, so I don't even have to worry about that!